### PR TITLE
[0.22] Backport of PRs 4815 5321 5444

### PR DESCRIPTION
**Pull request by cardil**
[0.22] Backport of PRs:

- knative#4815
- knative#5321
- knative#5444

Restoring manually modified vendor dir with: 481a26f

### CODE REVIEW

**Comment by cardil** [comment]
/hold

I'll try to verify this on serverless operator before merging

**Comment by skonto** [comment]
Hi what happened at the old vendor dir, I am missing a bit of the context here.

**Comment by cardil** [comment]
@skonto I don't understand your question. Could you rephrase it?

**Comment by skonto** [comment]
@cardil > Restoring manually modified vendor dir with: 481a26f
Why was modified manually?

**Comment by cardil** [comment]
@skonto There were some PRs like #1302 or #1282 that directly modifies the vendor directory without changing `go.mod` file. Then running `hack/update-deps.sh` scripts removes those patches. That's why I needed to restore them after cherry-picking my PRs.

See, a mailing thread on #serverless-dev about it: _"Adding CI checks on OpenShift knative forks"_

**Comment by cardil** [comment]
Here's a passing build on serverless operator with those changes: https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift-knative_serverless-operator/1011/pull-ci-openshift-knative-serverless-operator-main-4.7-upgrade-tests-aws-ocp-47/1402743138010796032

/unhold

Please review

/cc @matzew @mgencur

**Comment by matzew** [comment]
> Here's a passing build on serverless operator with those changes: https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift-knative_serverless-operator/1011/pull-ci-openshift-knative-serverless-operator-main-4.7-upgrade-tests-aws-ocp-47/1402743138010796032

Based on the above passed build for serverless, LGTM

/lgtm
